test/java/seedu/addressbook/commands/AddCommandTest.java

```diff
@@ -0,0 +1,159 @@
+package seedu.addressbook.commands;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.junit.Test;
+
+import seedu.addressbook.data.AddressBook;
+import seedu.addressbook.data.exception.IllegalValueException;
+import seedu.addressbook.data.person.Address;
+import seedu.addressbook.data.person.Email;
+import seedu.addressbook.data.person.Name;
+import seedu.addressbook.data.person.Person;
+import seedu.addressbook.data.person.Phone;
+import seedu.addressbook.data.person.ReadOnlyPerson;
+import seedu.addressbook.data.person.UniquePersonList;
+import seedu.addressbook.data.person.UniquePersonList.DuplicatePersonException;
+import seedu.addressbook.data.tag.UniqueTagList;
+
+public class AddCommandTest {
+    private static final List<ReadOnlyPerson> LAST_SHOWN_LIST = Collections.emptyList();
+    
+    @Test
+    public void addCommand_invalidName_throws() {
+        Set<String> tags = new HashSet<>();
```

Comment:
Maybe instead of constructing an empty `HashSet` every time for every test, maybe define a global `NO_TAGS` constant or something that is assigned the value of `Collections.emptySet()`.